### ISSUE TITLE
ChatParticipant#toString might yield NPE

### ISSUE THREAD

**Issue by RoiEXLab** [triplea-game/triplea#6136]
Just noticed this on a dev branch:
```
java.lang.NullPointerException
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:878)
	at org.triplea.domain.data.PlayerChatId.of(PlayerChatId.java:29)
	at org.triplea.domain.data.ChatParticipant.getPlayerChatId(ChatParticipant.java:43)
	at org.triplea.domain.data.ChatParticipant.toString(ChatParticipant.java:16)
	at java.base/java.lang.String.valueOf(String.java:2951)
	at java.base/java.util.Arrays.toString(Arrays.java:5113)
	at games.strategy.engine.message.RemoteMethodCall.classesToString(RemoteMethodCall.java:103)
	at games.strategy.engine.message.RemoteMethodCall.<init>(RemoteMethodCall.java:38)
	at games.strategy.engine.message.UnifiedInvocationHandler.invoke(UnifiedInvocationHandler.java:42)
	at com.sun.proxy.$Proxy6.speakerAdded(Unknown Source)
	at games.strategy.engine.chat.ChatController.joinChat(ChatController.java:113)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at games.strategy.engine.message.unifiedmessenger.EndPoint.invokeSingle(EndPoint.java:146)
	at games.strategy.engine.message.unifiedmessenger.EndPoint.invokeMultiple(EndPoint.java:129)
	at games.strategy.engine.message.unifiedmessenger.EndPoint.invokeLocal(EndPoint.java:114)
	at games.strategy.engine.message.unifiedmessenger.UnifiedMessenger.invokeAndWait(UnifiedMessenger.java:96)
	at games.strategy.engine.message.UnifiedInvocationHandler.invoke(UnifiedInvocationHandler.java:48)
	... 10 more
```
Apparently lombok invokes the `playerChatId` getter in toString which causes `PlayerChatId.of(playerChatId)` to throw an NPE if `playerChatId` is null.

Not sure how to really fix this issue. Just write our own toString, or make change the `PlayerChatId#`of method?

**Comment by DanVanAtta** [comment]
We might be encountering a modeling problem. Player-chat-id was not intended to be nullable, though I think wound up being that way to work with java socket chat. I'd suspect the correct fix here is to make the value non-nullable.

**Comment by RoiEXLab** [comment]
Sounds like the right thing to do, but it looks like there are a lot of places where this particular value is indeed not set and left null.

**Comment by DanVanAtta** [comment]
Should be fixed in #6237
